enhancements/proposals/2490-scylladbmonitoring-external-prometheus/README.md

```diff
@@ -0,0 +1,624 @@
+   }
+   ```
+
+   Strongly-typed enums will be used for defining a data source type and access mode:
+
+   ```go
+   // GrafanaDatasourceType defines the type of Grafana datasource.
+   // +kubebuilder:validation:Enum="Prometheus"
+   type GrafanaDatasourceType string
+
+   const (
+       // GrafanaDatasourceTypePrometheus is the Prometheus datasource type.
+       GrafanaDatasourceTypePrometheus GrafanaDatasourceType = "Prometheus"
+   )
+   ```
+
+   ```go
+   // GrafanaDatasourceAccess defines the access mode for Grafana datasource.
+   // +kubebuilder:validation:Enum="proxy";"direct"
+   type GrafanaDatasourceAccess string
+
+   const (
+       // GrafanaDatasourceAccessProxy means Grafana will proxy requests to the datasource.
+       GrafanaDatasourceAccessProxy GrafanaDatasourceAccess = "proxy"
+
+       // GrafanaDatasourceAccessDirect means Grafana will connect directly to the datasource from the browser.
+       GrafanaDatasourceAccessDirect GrafanaDatasourceAccess = "direct"
+   )
+   ```
+
+   `GrafanaPrometheusDatasourceSpec` will define Prometheus-specific options, including TLS and authentication.
+
+   ```go
+   type GrafanaPrometheusDatasourceSpec struct {
+      // tls holds TLS configuration for connecting to Prometheus over HTTPS.
+      // +optional
+      TLS *GrafanaDatasourceTLSSpec `json:"tls,omitempty"`
+
+      // auth holds authentication options for connecting to Prometheus.
+      // +optional
+      Auth *GrafanaPrometheusDatasourceAuthSpec `json:"auth,omitempty"`
+   }
+   ```
+
+   `GrafanaDatasourceTLSSpec` will define TLS configuration options for server certificate verification and client
+   authentication.
+
+   ```go
+   // LocalObjectKeySelector selects a key of a ConfigMap or Secret in the same namespace.
+   type LocalObjectKeySelector struct {
+      // name of the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Name string `json:"name"`
+
+      // key within the selected object.
+      // +kubebuilder:validation:MinLength=1
+      Key string `json:"key"`
+   }
+
+   type GrafanaDatasourceTLSSpec struct {
+      // caCert is a reference to a key within the CA bundle ConfigMap. The key should hold the CA cert in PEM format.
+      // When not specified, system CAs are used.
+      // +optional
+      CACertConfigMapRef *LocalObjectKeySelector `json:"caCertConfigMapRef,omitempty"`
+
+      // insecureSkipVerify controls whether to skip server certificate verification.
+      // +kubebuilder:default:=false
+      // +optional
+      InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+
+	  // clientTLSKeyPairSecretName is a name of a Secret holding client TLS certificate and key for mTLS authentication.
+	  // It's expected to be a standard Kubernetes TLS Secret with `tls.crt` and `tls.key` keys.
+      // +optional
+      ClientTLSKeyPairSecretName string `json:"clientTLSKeyPairSecretName,omitempty"`
```

Comment:
nit: doesn't the name go against https://github.com/kubernetes/community/blob/d3c6c06c43b882c8aca692dd92e1adedad1dfab8/contributors/devel/sig-architecture/api-conventions.md#naming-of-the-reference-field?

Reply:
Good point. Let's also go with `corev1.LocalObjectReference` for the reference type instead of a string.